src/hashes/hasher.ts

```diff
@@ -3,8 +3,35 @@ import type { MultihashHasher } from './interface.js'
 
 type Await<T> = Promise<T> | T
 
-export function from <Name extends string, Code extends number> ({ name, code, encode }: { name: Name, code: Code, encode(input: Uint8Array): Await<Uint8Array> }): Hasher<Name, Code> {
-  return new Hasher(name, code, encode)
+const DEFAULT_MIN_DIGEST_LENGTH = 20
+const DEFAULT_MAX_DIGEST_LENGTH = 128
+
+export interface HasherInit <Name extends string, Code extends number> {
+  name: Name
+  code: Code
+  encode(input: Uint8Array): Await<Uint8Array>
+
+  /**
+   * The minimum length a hash is allowed to be in bytes
+   *
+   * @default 20
+   */
+  minDigestLength?: number
+
+  /**
+   * The maximum length a hash is allowed to be in bytes
+   *
+   * @default 128
+   */
+  maxDigestLength?: number
+}
+
+export function from <Name extends string, Code extends number> ({ name, code, encode, minDigestLength, maxDigestLength }: HasherInit<Name, Code>): Hasher<Name, Code> {
+  return new Hasher(name, code, encode, minDigestLength, maxDigestLength)
+}
+
+export interface DigestOptions {
```

Comment:
I'd add the unit type in the JS doc so it's clear it's in bytes.

Reply:
I've added a doc comment, let me know if you think it's clear.